async/kafka/kafka.go

```diff
@@ -39,6 +38,19 @@ func init() {
 	prometheus.MustRegister(topicPartitionOffsetDiff)
 }
 
+// ConsumerConfig is the common configuration of patron kafka consumers
+type ConsumerConfig struct {
+	Brokers      []string
+	Buffer       int
+	DecoderFunc  encoding.DecodeRawFunc
+	SaramaConfig *sarama.Config
+}
+
+//Consumer interface exports the two config objects in order to be modified by option functions.
+type Consumer interface {
+	ConsumerConfig() *ConsumerConfig
+}
+
 type message struct {
```

Comment:
Maybe we should split the message also inside the packages...
simple message does not need the session

Reply:
Yes we could. But I think then we will have just some functions (determineDecoder, getCorrelationID, determineContentType, mapHeader) in common code in `kafka.go`.

And they should be public in order to be used in each package's `claimMessage`. At least I haven't thought anything else :confused:

What do you think?

Reply:
You are right, let's leave it as is...